aya/src/programs/sock_ops.rs

```diff
@@ -58,9 +72,16 @@ impl SockOps {
     ///
     /// The returned value can be used to detach, see [SockOps::detach].
     pub fn attach<T: AsFd>(&mut self, cgroup: T) -> Result<SockOpsLinkId, ProgramError> {
+        self.attach_with_flags(cgroup, SockOpsFlags::empty())
+    }
+
+    /// Attaches the program to the given cgroup.
+    ///
+    /// The returned value can be used to detach, see [SockOps::detach].
+    pub fn attach_with_flags<T: AsFd>(&mut self, cgroup: T, flags: SockOpsFlags) -> Result<SockOpsLinkId, ProgramError> {
         let prog_fd = self.fd()?;
 
-        let link = ProgAttachLink::attach(prog_fd.as_fd(), cgroup.as_fd(), BPF_CGROUP_SOCK_OPS)?;
+        let link = ProgAttachLink::attach_with_flags(prog_fd.as_fd(), cgroup.as_fd(), BPF_CGROUP_SOCK_OPS, flags.bits())?;
```

Comment:
Aside: Given all the other Cgroup programs can be attached with bpf_link - see: https://github.com/aya-rs/aya/blob/d5414bf10c80ae8cef757f0cdf06bfdd38746daa/aya/src/programs/cgroup_device.rs#L69

I'd wager that this is likely the case for sock_ops programs too. I'll open an issue for that.

Reply:
👍